pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -117,6 +117,15 @@ func (fr *FlowRecords) ValidateAndUpdateStats(connKey flowexporter.ConnectionKey
 
 }
 
+// SetDyingAndDoneExport sets the appropriate field of the record to true given
+// the connection key and record. It also updates the record map. Caller is expected
+// to grab lock to the record map.
+func (fr *FlowRecords) SetDyingAndDoneExport(connKey flowexporter.ConnectionKey, record flowexporter.FlowRecord) {
+	record.DyingAndDoneExport = true
+
+	fr.recordsMap[connKey] = record
+}
+
```

Comment:
I find this function a bit confusing, and the name is not really helping. Since you only use it in one place, do you think you could inline the code instead, and not define this function at all?

Reply:
Tried to do that initially but could not as the flow record map is not public. I modified it in a different way. Please check.